wagtail/users/tests/test_admin_views.py

```diff
@@ -2041,7 +2056,13 @@ def flatten(perm_set):
                     perm.content_type.model,
                 )
                 for perm_set in object_perms
-                for perm in [next(v for v in flatten(perm_set) if "perm" in v)["perm"]]
```

Comment:
The `if "perm" in v` in this original line was done to check if it's a dictionary that contains the `perm` key, but due to how the `flatten` method above is implemented, `v` can also be a string. This means the `if` check would also pass for strings that contain `perm`.

After adding the custom permissions in the test model, which contains the substring `"perm"`, accessing the `["perm"]` key would then crash with `TypeError: string indices must be integers`. Hence I added `if isinstance(v, dict)` to filter out the non-dict values.